### ISSUE TITLE
Warm product cache bypasses signature validation

### ISSUE THREAD

**Issue by Concode0** [Concode0/Versor#25]
### What happened?

Equal-dimensional algebras run independently, but a warmed product cache can accept explicit compact layouts from a different signature and use the wrong metric coefficients.

### Where?

clifra.core.planning.GradePlanner.product_executor_for_layouts

### Algebra signature

Cl(3,0,0) target; Cl(0,3,0) supplied layouts

### Minimal reproduction

```python
import torch
from clifra import make_algebra
from clifra.core.runtime.tensors import LaneStorage

target = make_algebra(3, 0, 0, device="cpu")
foreign = make_algebra(0, 3, 0, device="cpu")
x = torch.tensor([[1.0, 0.0, 0.0]])

def square(algebra, source):
    return algebra.projected_product(
        x, x,
        left_layout=source.layout((1,)),
        right_layout=source.layout((1,)),
        output_layout=source.layout((0,)),
        left_storage=LaneStorage.COMPACT,
        right_storage=LaneStorage.COMPACT,
    )

square(target, target)       # warm target cache
print(square(target, foreign))  # tensor([[1.]]) instead of rejecting
print(square(foreign, foreign)) # tensor([[-1.]])
```

### Error output

```shell
square(target, target)       # warm target cache
print(square(target, foreign))  # tensor([[1.]]) instead of rejecting
print(square(foreign, foreign)) # tensor([[-1.]])

# Correct works should
# Reject the mismatched signature before consulting the executor cache,
# consistently in both cold- and warm-cache states.
```

**Comment by Concode0** [comment]
**Solutions currently under consideration**

Rigorously validate the supplied layouts/TensorContracts against the receiving AlgebraSpec before cache lookup. No instance-ownership mechanism is needed. Apply the same pre-cache contract check to plan_product and add cold- and warm-cache regression tests.
